locator/tablets.cc

```diff
@@ -1303,12 +1304,18 @@ static void assert_rf_rack_valid_keyspace(std::string_view ks, const token_metad
         tablet_logger.debug("[assert_rf_rack_valid_keyspace]: Verifying for '{}' / '{}'", ks, dc);
 
         size_t normal_rack_count = 0;
+        size_t transitioning_rack_count = 0;
         for (const auto& [_, rack_nodes] : rack_map) {
             // We must ignore zero-token nodes because they don't take part in replication.
             // Verify that this rack has at least one normal node.
             const bool normal_rack = std::ranges::any_of(rack_nodes, is_normal_token_owner);
             if (normal_rack) {
                 ++normal_rack_count;
+            } else {
+                const bool is_transitioning_rack = std::ranges::any_of(rack_nodes, is_transitioning_token_owner);
+                if (is_transitioning_rack) {
+                    ++transitioning_rack_count;
+                }
```

Comment:
Looks like the rack is considered to be transitioning if _any_ of the nodes in the rack are boostrapping/leaving. However, later you write:

```
        // If some rack is undetermined because it has only transitioning nodes and no normal nodes, we cannot
        // use normal_rack_count to determine RF-rack-validity.
```

I agree that the rack should be undetermined if _all_ its nodes are transitioning - but it looks like the implementation does not follow this. It looks like a bug in the code.

Reply:
Oh, right, it's in the `else` branch.